presto-main/src/main/java/com/facebook/presto/execution/buffer/PageCodecMarker.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.buffer;
+
+public enum PageCodecMarker
+{
+    COMPRESSED(1);
+
+    private final int mask;
+
+    PageCodecMarker(int flagBitPosition)
+    {
+        if (flagBitPosition <= 0 || flagBitPosition > 8) {
+            throw new IllegalArgumentException("PageCodecMarker flagBitPosition must be between 1 and 8. Found: " + flagBitPosition);
```

Comment:
use checkArgument: `checkArgument(bit >= 1 && bit <= 8, "bit must be between 1 and 8; found " + bit);`

Reply:
Will change to use `checkArgument `